src/lib/yang/binary.lua

```diff
@@ -140,9 +140,17 @@ local function data_emitter(production)
       end
    end
    function handlers.struct(production)
-      local member_names = {}
-      for k,_ in pairs(production.members) do table.insert(member_names, k) end
-      table.sort(member_names)
+      local member_keys = {}
+      for k,_ in pairs(production.members) do table.insert(member_keys, k) end
+      local function order_predicate (x, y)
+         if (type(x) == 'number' and type(y) == 'number')
+         or (type(x) == 'string' and type(y) == 'string') then
```

Comment:
I usually write it in the former way (I think because lua-mode auto-indents it nicely that way?) but I find both versions readable. I don’t mind either way. :+1: